src/Symfony/Component/Uid/UuidV6.php

```diff
@@ -43,6 +43,33 @@ public function getNode(): string
         return substr($this->uid, 24);
     }
 
+    public function toV7(): UuidV7
+    {
+        $uuid = $this->uid;
+        $time = BinaryUtil::hexToNumericString('0'.substr($uuid, 0, 8).substr($uuid, 9, 4).substr($uuid, 15, 3));
+        if ('-' === $time[0]) {
+            throw new \InvalidArgumentException('Cannot convert UUID to v7: its timestamp is before the Unix epoch.');
+        }
+
+        $ms = \strlen($time) > 4 ? substr($time, 0, -4) : '0';
+        $time = dechex(10000 * hexdec(substr($uuid, 20, 3)) + substr($time, -4));
+
+        if (\strlen($time) > 6) {
+            $uuid[29] = $uuid[29] ^ chr($time[0]);
```

Comment:
There's an "overflow" here, for example, if `$uuid[29] === 'f'` and `$time[0] === '1'`, xoring them gives `g` which is invalid. I added a failing test for it :smiley:

Reply:
indeed, fixed!